coverage/docs/design.md

```diff
@@ -0,0 +1,139 @@
+#Overview
+This code coverage tool has two major features.
+1. In post-submit workflow (when a pull request is merged), it runs code coverage and generates the 
+  following artifacts in a GCS bucket
+  - xml file that stores file-level and package-level code coverage, formatted to be readable by TestGrid
+  - code coverage profile, which will be used later in presubmit workflow as a base for comparison
+2. In pre-submit workflow (when a pull requested is created or updated with new commit)
+  - it runs code coverage on target directories and compares the new result with the one stored by 
+  the post-submit workflow and generate coverage difference. It reports coverage changes to the 
+  pull request as a comment by a robot github account. 
+  - it uses go tools to generate line by line coverage and serve the result in html, with a link as 
+  part of the robot comment mentioned above 
+  - it can report a failure status if coverage falls below threshold - this allows repo owner to make the 
```

Comment:
Your job does not do this. This is done by Prow's `plank` component. I think what you mean to say is that the job can be configured to exit with a non-zero status if the line coverage falls below the configured threshold.

Reply:
Done.